src/parsing/internal-parsing-logic.ts

```diff
@@ -221,10 +259,22 @@ function convertASTToInstructionSet(retVal: any[], mem: any[], expression: any[]
                 } else {
                     retVal.push(expression[0])
                 }
-                retVal.push(mem[mem.length - 2])
-                retVal.push(mem[mem.length - 1])
-                mem.pop()
-                mem.pop()
+                var not = false
+                if( expression[0].includes("PLA")) {
+                    var it = parseInt(expression[0].replace("PLA", "").trim())
+                    if(originalDelimiters[it] == 'NOT') {
+                        not = true
+                    }
+                }
+                if(not) {
+                    retVal.push(mem[mem.length - 1])
+                    mem.pop()
+                } else {
+                    retVal.push(mem[mem.length - 2])
```

Comment:
Could this and next 4 lines be combined in `retVal.push(mem.splice(mem.length - 2, 2))`?